x-pack/plugins/reporting/server/lib/content_stream.ts

```diff
@@ -265,22 +311,26 @@ export class ContentStream extends Duplex {
     }
 
     this.bytesWritten += chunk.byteLength;
-    this.buffer = this.buffer.slice(size);
+
+    this.buffers = partiallyFlushedBuffer
+      ? [partiallyFlushedBuffer, ...this.buffers.slice(buffersToFlush.length)]
+      : this.buffers.slice(Math.max(buffersToFlush.length - 1, 1));
+
+    this.bytesBuffered -= bytesToFlush;
   }
 
   private async flushAllFullChunks() {
     const maxChunkSize = await this.getMaxChunkSize();
 
-    while (this.buffer.byteLength >= maxChunkSize) {
+    while (this.bytesBuffered >= maxChunkSize && this.buffers.length) {
```

Comment:
The second check is redundant since the `bytesBuffered` must be positive, and in this case, the `buffers` array will contain at least one element.
```suggestion
    while (this.bytesBuffered >= maxChunkSize) {
```

Reply:
That's a good point!